internal/infrastructure/storage/db/badger/trade_repository_impl.go

```diff
@@ -99,6 +99,8 @@ func (t tradeRepositoryImpl) UpdateTrade(
 	ID *uuid.UUID,
 	updateFn func(t *domain.Trade) (*domain.Trade, error),
 ) error {
+	isManaged := ctx.Value("tx") == nil
```

Comment:
isManaged -> means someone outside manages tx and that it should be:
`	isManaged := ctx.Value("tx") != nil
`

Reply:
ok I have a different "semantic" in mind :) I mean managed from an external point of view: the repository method "manages" the creation, commit, rollback and retry for you, without caring to give us.

Maybe better calling `txIsGiven`

Reply:
done